stock_picking_backorder_reason/models/stock_picking_type.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockPickingType(models.Model):
+
+    _inherit = "stock.picking.type"
+
+    backorder_reason = fields.Boolean(
+        help="Check this if you want people selecting a backorder reason when applicable."
+        "This will trigger a backorder strategy rule defined on that reason."
+    )
+    backorder_reason_sale = fields.Boolean(
+        compute="_compute_backorder_reason_sale",
+        readonly=False,
+        store=True,
+        help="Check this in order to consider pickings in this type as Sale ones for "
+        "backorder reason cancellation.",
+    )
+    backorder_reason_purchase = fields.Boolean(
+        compute="_compute_backorder_reason_purchase",
+        readonly=False,
+        store=True,
+        help="Check this in order to consider pickings in this type as "
+        "Purchase ones for backorder reason cancellation.",
+    )
+
+    @api.depends("backorder_reason")
+    def _compute_backorder_reason_sale(self):
+        activated_types = self.filtered("backorder_reason_sale")
+        to_deactivate = activated_types.filtered(lambda t: not t.backorder_reason)
```

Comment:
a double loop.... should be done in one loop.

Reply:
Done